test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/SourcesCommandTests.cs

```diff
@@ -0,0 +1,356 @@
+                        });
+                    });
+
+                mockServer.Get.Add(
+                    builder.GetRepoSignIndexPath(),
+                    request =>
+                    {
+                        return new Action<HttpListenerResponse>(response =>
+                        {
+                            var mockResponse = builder.BuildRepoSignIndexResponse();
+                            response.ContentType = mockResponse.ContentType;
+                            MockServer.SetResponseContent(response, mockResponse.Content);
+                        });
+                    });
+
+                Util.CreateFile(configFileDirectory, configFileName,
+                    $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <packageSources>
+    <add key=""test_source"" value=""{builder.GetV3Source()}"" />
+  </packageSources>
+  <trustedSources>
+    <test_source>
+      <add key=""3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece"" value=""CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"" fingerprintAlgorithm=""SHA256"" />
+    </test_source>
+  </trustedSources>
+</configuration>");
+
+                var args = new string[] {
+                    "sources",
+                    "update",
+                    "-Name",
+                    "test_source",
+                    "-Source",
+                    "http://test_source",
+                    "-ConfigFile",
+                    configFilePath
+                };
+
+                var root = Directory.GetDirectoryRoot(Directory.GetCurrentDirectory());
+
+                mockServer.Start();
+
+                // Act
+                var result = CommandRunner.Run(nugetexe, root, string.Join(" ", args), true);
+
+                // Assert
+                mockServer.Stop();
+
+                Assert.Equal(0, result.Item1);
+                var settings = Settings.LoadDefaultSettings(configFileDirectory, configFileName, null);
+                var source = new PackageSourceProvider(settings).LoadPackageSources().Where(p => p.Name == "test_source").FirstOrDefault();
+
+                Assert.Equal("http://test_source", source.Source);
+                Assert.NotNull(source.TrustedSource);
+                Assert.Null(source.TrustedSource.ServiceIndex);
+                Assert.Equal(1, source.TrustedSource.Certificates.Count);
+
+                var cert = source.TrustedSource.Certificates.FirstOrDefault();
+
+                Assert.Equal("3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece", cert.Fingerprint);
+                Assert.Equal(HashAlgorithmName.SHA256, cert.FingerprintAlgorithm);
+                Assert.Equal("CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", cert.SubjectName);
+            }
+        }
+
+        [Fact]
+        public void SourcesCommandTest_AddTrustedSourceUnsupport()
```

Comment:
Can you rename these tests in the form:  _TestTarget_Condition_Expectation_

The class name is SourcesCommandTest, so adding it to every method name is redundant.  A better name for this test which clearly describes what's being tested and the expectation is:

    Add_WithTrustAndUnsupportedPackageSource_Fails